Client/Frontend/Browser/TabManager.swift

```diff
@@ -40,6 +40,8 @@ class WeakTabManagerDelegate {
 class TabManager: NSObject {
     private var delegates = [WeakTabManagerDelegate]()
     weak var stateDelegate: TabManagerStateDelegate?
+    var BrowserObject : BrowserViewController?
```

Comment:
Can you try to fix this without adding a reference to the browserviewcontroller? Coupling things like this will probably make understanding the relationship between these 2 classes more difficult as time passes and more code is added. 

Try looking at how a home panel is loaded inside BVC. Maybe you can add logic there to select the urlbar when no homepanel is provided.